contrib/babelfishpg_tds/src/backend/tds/tdsutils.c

```diff
@@ -1148,3 +1186,113 @@ is_babelfish_ownership_enabled(ArrayType *array)
 	}
 	return false;
 }
+
+/*
+ * handle_alter_role
+ *
+ * Description: This function handles dealing with ALTER ROLE <role> WITH.
+ *
+ * Returns: true - We're not attempting to modify something we shouldn't have access to. Normal security checks.
+ *          false - We've reported an error and should not continue executing this call.
+ */
+static bool
+handle_alter_role(AlterRoleStmt* alter_role_stmt)
+{
+	List *options = alter_role_stmt->options;
+	ListCell *opt;
+	char *name = get_role_name(alter_role_stmt->role);
+	bool allow_alter_operation = enable_alter_babelfish_role;
+
+	/*
+	 * If the role does not exist, just let the normal Postgres checks happen.
+	 * If the role is "sysadmin" then allow the alter operation.
+	 */
+	if (name == NULL || (strcmp(name, BABELFISH_SYSADMIN) == 0))
```

Comment:
Please check how we need to handle role which is a member of "sysadmin"

Reply:
+1

Reply:
In the function `is_babelfish_role` there is a check whether the given role is member of `sysadmin` and if it is present then will throw an error saying you cannot alter the role.

Reply:
We are using `is_babelfish_role` after the `for` loop. We can skip the for loop, instead we can directly return true as we are doing for `sysadmin`.